tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -893,6 +893,44 @@ class SerializableModelLibraryGenerator {
 
     Expression fieldExpression = fieldRef;
 
+    // If the field is a custom class and we are generating 'toJsonForProtocol',
+    // we need to check if it implements the ProtocolSerialization interface. If
+    // it does, we use the 'toJsonForProtocol' method to convert the field to
+    // JSON. Otherwise we use the 'toJson' method.
+    if (fieldType.customClass && toJsonMethodName == 'toJsonForProtocol') {
```

Comment:
Sugggestion: Create enums (or constant variables) for `'toJsonForProtocol'` so that we only have a single definition of it.